internal/github/github.go

```diff
@@ -128,3 +129,106 @@ func (c *GHClient) GetCIStatus(ctx context.Context, owner, repoName string, ref
 
 	return CIStatusUnknown, nil
 }
+
+func (c *GHClient) getOneStatus(ctx context.Context, owner, repoName, ref, check string) (CIStatus, error) {
+	listOptions := github.ListOptions{
+		PerPage: 100,
+	}
+
+	opt := &github.ListCheckRunsOptions{
+		CheckName:   github.String(check),
+		ListOptions: listOptions,
+		Filter:      github.String("latest"),
+	}
+
+	checkRuns, _, err := c.client.Checks.ListCheckRunsForRef(ctx, owner, repoName, ref, opt)
```

Comment:
Suggestion, maybe something to consider for a follow-up (as it could apply to multiple locations in the codebase). Using `github.CheckResponse(...)` to validate the response for errors (check things like unauthorised, rate-limited, etc.)

```suggestion
	checkRuns, resp, err := c.client.Checks.ListCheckRunsForRef(ctx, owner, repoName, ref, opt)
	if err := github.CheckResponse(resp.Response); err != nil {
		// process the error(s)
	}
```